operatorcourier/validate.py

```diff
@@ -91,7 +95,7 @@ def _bundle_validation(self, bundle, repository=None):
 
         return valid
 
-    def _crd_validation(self, bundleData):
+    def _crd_validation(self, bundleData, nested):
```

Comment:
Why are we passing this variable along to each of these methods and then not using it for anything?

Reply:
Because we need `nested` value for `_pkgs_validation`, which is one of the [`validators`](https://github.com/jeremylinlin/operator-courier/blob/support-nested-directory-layout/operatorcourier/validate.py#L362). By changing the function signature of this, we also need to update the other ones, otherwise errors will be thrown.

Reply:
This is definitely a code smell. Passing these functions along without a way to abstract the function header seems like bad design. If we genuinely have different validation depending on whether or not a package is nested, we should have some way of abstracting that at the top level -- perhaps by having different root methods that call the same underlying validation methods on certain subfields. Otherwise this will become extremely difficult to maintain as we get new validation needs.